### DISCUSSION TITLE
How to control the Die to Core row offset during floorplan

### DISCUSSION THREAD

**Discussion by gkamendje** [The-OpenROAD-Project/OpenROAD-flow-scripts#1551] [Q&A]
I have the following lines 
```
export CORE_UTILIZATION = 20 
export CORE_ASPECT_RATIO = 1
```
in my design `config.mk` file and I see that the tool always places the first row at an offset equivalent to the height of a placement site (4.480 in the example on the attached picture)
![RowOffset](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/56942214/f9b9035e-9edb-4151-a247-ce07e8b42329). 
I was wondering if there is a way to force to the first row to start at an offset which is not a multiple of the height of the placement site. I have tried things like these without success
```
DIE_AREA    =  0   0   684.88 135.5
CORE_AREA   =  1.12   1.96  683.76 126.02 
```
Using the `ROW` directive in a `.def` file, it is possible to let the rows start anywhere. Is there a way to achieve this in a `config.mk` file? 

More generally, is there an issue starting the rows at an offset which is not a multiple of the height of the placement site?

**Comment by maliberty** [comment]
If you use CORE_UTILIZATION / CORE_ASPECT_RATIO  you can also specify CORE_MARGIN.  In either case we do snap to a site multiple, you should get a message like ```Core area lower left ... snapped to ...```.   

I'm not sure of the original motivation for snapping.  We could remove the snapping though I'm not sure if other problems will turn up elsewhere.  I do worry that putting the core very close to the edge could lead to routing issues with the IO pins.  Would you say more about your use model?

**Reply by gkamendje** [reply]
The use model is a block with very few pins (with respect to the perimeter). The primary goal is the minimization of the overall area. Snapping to the site multiple is quite an overkill in this context since there are very few nets to the routed in that region. I could think of a warning being issued to notify that the first row is too close to the edge while obeying the `CORE_AREA` directive.

**Reply by maliberty** [reply]
There is generally an intentional design between the track pattern and the expected site locations.  If we allow arbitrary locations I expect we will see a lot of routing issues.

Is saving ~3um a major concern?  How small is this block?

**Reply by gkamendje** [reply]
The block is ~300x150um. Manually specifying the rows via a DEF file is good enough for now and I do not see any major routing issue. Of course this might be a totally different story as the pin count increases.

**Reply by maliberty** [reply]
If you are ok with that solution I think it is preferrable to keep the snapping to avoid problems in general.
